content/blog/announcing-pulumi-preview-improvments/index.md

```diff
@@ -0,0 +1,72 @@
+---
+title: "Announcing Pulumi Preview Improvements"
+date: 2020-10-27
+draft: false
+meta_desc: "Announcing a significant improvement to the pulumi preview experience."
+#meta_image: 
+authors:
+    - paul-stack
+tags:
+    - enhancement
+    - pulumi
+---
+
+Today we are announcing a minor but significant improvement to the Pulumi [preview]({{< relref "/docs/reference/cli/pulumi_preview" >}}
+experience.
+
+<!--more-->
+
+## Change in Behaviour
+
+We can look at a sample Pulumi application that creates a Google GKE cluster. Pulumi then deploys an application into the
+Kubernetes cluster.
+
+{{< gist pulumipus 0c6e4ede557600e114885869ea0aeb96 >}}
+
+A simple change to the labels for the GKE cluster led to a misleading preview experience that told users that the Kubernetes
+resources were going to be replaced:
+
+```
+     Type                              Name                         Plan        Info
+     pulumi:pulumi:Stack               gcp-ts-gke-hello-world-dev2
+ ~   ├─ gcp:container:Cluster          helloworld                   update      [diff: ~resourceLabels]
+ +-  ├─ gcp:container:NodePool         primary-node-pool            replace     [diff: ~location]
+ +-  ├─ pulumi:providers:kubernetes    helloworld                   replace     [diff: ~kubeconfig]
+ +-  ├─ kubernetes:core/v1:Namespace   helloworld                   replace     [diff: -metadata~provider]
+ +-  ├─ kubernetes:core/v1:Service     helloworld                   replace     [diff: ~metadata,provider]
+ +-  └─ kubernetes:apps/v1:Deployment  helloworld                   replace     [diff: ~metadata,provider]
+
+Resources:
+    ~ 1 to update
+    +-5 to replace
+    6 changes. 1 unchanged
+```
+
+The result of the Pulumi apply showed that only the `gcp:container:Cluster` changed, and no other Kubernetes resources were replaced.
+
+```
+     Type                      Name                         Status      Info
+     pulumi:pulumi:Stack       gcp-ts-gke-hello-world-dev2
+ ~   └─ gcp:container:Cluster  helloworld                   updated     [diff: ~resourceLabels]
+
+Resources:
+    ~ 1 updated
+    6 unchanged
+```
+
+This change was what we expected to happen. The new preview functionality ensures a correct output on preview:
+
+```
+     Type                      Name                         Plan       Info
+     pulumi:pulumi:Stack       gcp-ts-gke-hello-world-dev2
+ ~   └─ gcp:container:Cluster  helloworld                   update     [diff: ~resourceLabels]
+
+Resources:
+    ~ 1 to update
+    6 unchanged
+```
+
+## Conclusion
```

Comment:
Possibly worth also mentioning the scenario in https://github.com/pulumi/pulumi-azure/issues/192?